lib/screens/explore_screens/vote_screens/proposals/viewmodels/proposal_view_model.dart

```diff
@@ -0,0 +1,143 @@
+import 'dart:io';
+
+import 'package:intl/intl.dart';
+import 'package:seeds/datasource/remote/model/proposal_model.dart';
+import 'package:seeds/datasource/remote/model/referendum_model.dart';
+import 'package:timeago/timeago.dart' as timeago;
+
+class ProposalViewModel {
+  final int id;
+  final int settingValue;
+  final String settingName;
+  final String creator;
+  final String recipient;
+  final String quantity;
+  final int total;
+  final int favour;
+  final int against;
+  final String title;
+  final String summary;
+  final String description;
+  final String image;
+  final String url;
+  final String status;
+  final String stage;
+  final int creationDate;
+  final String campaignType;
+  final int voiceNeeded;
+
+  ProposalViewModel({
+    required this.id,
+    required this.settingValue,
+    required this.settingName,
+    required this.creator,
+    required this.recipient,
+    required this.quantity,
+    required this.total,
+    required this.favour,
+    required this.against,
+    required this.title,
+    required this.summary,
+    required this.description,
+    required this.image,
+    required this.url,
+    required this.status,
+    required this.stage,
+    required this.creationDate,
+    required this.campaignType,
+    this.voiceNeeded = 0,
+  });
+
+  /// Percentage to advance 0-1 scale
+  double get voiceNeededBarPercent => ((voiceNeeded * 100) / total) / 100;
+
+  /// Percentage in favour 0-1 scale
+  double get favourAgainstBarPercent => total == 0 ? 0 : (favour.toDouble() / total.toDouble());
+
+  String get favourPercent => total > 0 ? '${((favour * 100) / total).toStringAsFixed(0)} %' : '0 %';
+
+  String get againstPercent => total > 0 ? '${((against * 100) / total).toStringAsFixed(0)} %' : '0 %';
+
+  String get createdAt {
+    final created = DateTime.fromMillisecondsSinceEpoch(creationDate * 1000);
+    if (DateTime.now().difference(created) > const Duration(days: 7)) {
+      return DateFormat.yMd(Platform.localeName.split('_').first).format(created);
+    } else {
+      return timeago.format(created);
+    }
+  }
+
+  String get categoryTypeLabel {
+    return campaignType == 'cmp.funding' || campaignType == 'cmp.invite' ? 'campaign' : campaignType;
+  }
+
+  ProposalViewModel copyWith(int voiceNeeded) {
+    return ProposalViewModel(
+      id: id,
+      settingValue: settingValue,
+      settingName: settingName,
+      creator: creator,
+      recipient: recipient,
+      quantity: quantity,
+      total: total,
+      favour: favour,
+      against: against,
+      title: title,
+      summary: summary,
+      description: description,
+      image: image,
+      url: url,
+      status: status,
+      stage: stage,
+      creationDate: creationDate,
+      campaignType: campaignType,
+      voiceNeeded: voiceNeeded,
+    );
+  }
+
+  factory ProposalViewModel.fromProposal(ProposalModel proposal) {
+    return ProposalViewModel(
```

Comment:
Maybe this is just me but - I don't understand the benefits of copying data like that.

There's one argument saying hey view models must only deal with the views, and I get that

But the other argument is DRY - don't repeat yourself, and in my experience with coding DRY is a higher rule. 

Practically speaking, if we change anything in proposals we need to change all this code too, and this is the reason we end up with > 500 line changes sometimes, because all values are copied all over the place. 

@gguijarro-c-chwy would like to have your opinion

This is more a generic observation. Not a blocker for the PR. 

Copying a ton of code makes me cringe.